demo/blocks_update/run.py

```diff
@@ -1,17 +1,44 @@
-from gradio import Dropdown, Blocks, Slider
+import gradio as gr
 
+with gr.Blocks() as block:
+    gr.Markdown(
+        """
+    # Animal Generator
+    Once you select a species, the detail panel should be visible.
+    """
+    )
 
-def update_dropdown_choices(val):
-    if val < 50:
-        opts = ["fail", "repeat"]
-    else:
-        opts = ["succeed", "repeat"]
-    return Dropdown.update(choices=opts)
+    species = gr.Radio(label="Animal Class", choices=["Mammal", "Fish", "Bird"])
+    animal = gr.Dropdown(label="Animal", choices=[])
 
+    with gr.Column(visible=False) as details_col:
+        weight = gr.Slider(0, minimum=0, maximum=20)
+        details = gr.Textbox(label="Extra Details")
+        generate_btn = gr.Button("Generate")
+        output = gr.Textbox(label="Output")
 
-with Blocks() as block:
-    s = Slider(default_value=0, minimum=0, maximum=100)
-    d = Dropdown(choices=["fake", "choices", "exist"])
+    species_map = {
+        "Mammal": ["Elephant", "Giraffe", "Hamster"],
+        "Fish": ["Shark", "Salmon", "Tuna"],
+        "Bird": ["Chicken", "Eagle", "Hawk"],
+    }
 
-    s.change(fn=update_dropdown_choices, inputs=[s], outputs=[d])
-    block.launch()
+    def filter_species(species):
+        return gr.Dropdown.update(
+            choices=species_map[species], value=species_map[species][1]
+        ), gr.update(visible=True)
+
+    species.change(filter_species, species, [animal, details_col])
+
+    def filter_weight(animal):
+        if animal in ("Elephant", "Shark", "Giraffe"):
+            return gr.update(maximum=100)
+        else:
+            return gr.update(maximum=20)
```

Comment:
I am not very fond of `gr.update` and I don't think it is intutive. WDYT of just having `gr.Dropdown.update` syntax? This would decrease the error occurences, and would make the code more intuitive. Furthermore it would make following/respecting component's field names easier.

Reply:
I think it's nice for a quick shortcut. It will fail if any invalid argument is passed to it because gr.update passes to gr.Dropdown.update right away.